### PR TITLE
Canvas: Update zoom dropdown to align with help center

### LINKED ISSUE #6912: Update zoom dropdown to align with help center
## Feature Description

This ticket should:

- [ ] Update the dropdown for zoom to use the new design system dropdown
- [ ] Make sure the design corresponds to the Figma guide
- [ ] Make sure the zoom dropdown and help center toggle are perfectly aligned as in Figma
- [ ] Make sure that there are no references to `DEPRECATED_THEME`
---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by samwhale**
## Context

Using the design system in the editor. No styles in the carousel use the deprecated theme.

image
[Designs](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=274%3A250)

Chat with designers about the chevron: https://xwp.slack.com/archives/C01ESBB8ZN1/p1617050704003700

Tasks:
- [x] Update the dropdown for zoom to use the new design system dropdown
- [x] Make sure the design corresponds to the Figma guide
- [x] Make sure the zoom dropdown and help center toggle are perfectly aligned as in Figma
- [x] Make sure that there are no references to `DEPRECATED_THEME`

## Summary

Updated things:
- Zoom dropdown uses the `DropDown` from the design system

## Relevant Technical Choices

- Added a `showChevron` boolean to give us the ability to hide the chevron like in the designs

## To-do

- [x] talk with design about whether to keep checkmark in the menu or not
- [ ] fix checkmark icon for dropdown in RTL (not related to this pr)

## User-facing changes

|Before|After|After RTL|
|--|--|--|
|image|image|image|

## Testing Instructions

1. Make sure `Canvas Zoom` experiment is turned on
2. Go to editor
3. See the zoom dropdown in the lower right hand corner of the canvas
4. Play with it - should be keyboard accessible too

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Make sure `Canvas Zoom` experiment is turned on
2. Go to editor
3. See the zoom dropdown in the lower right hand corner of the canvas
4. Play with it - should be keyboard accessible too

### UAT

- [x] UAT should use the same steps as above.

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6912

### CODE REVIEW

File: assets/src/edit-story/components/carousel/zoomSelector/dropdownItem.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}
+    >
+      {option.label}
+    </ListItemDisplayText>
+  </ListItem>
+);
+
+export default forwardRef(DropDownItem);
+
+DropDownItem.displayName = 'DropDownItem';
```
Comment (maxyinger) [inline assets/src/edit-story/components/carousel/zoomSelector/dropdownItem.js]: oh this is nice. I've just been doing this even tho it felt gross:
```
const SomeComponent = forwardRef(function SomeComponent(props, ref) { ... })
```

**Review by maxyinger — APPROVED** [review]
Looks good!

File: assets/src/edit-story/components/carousel/zoomSelector/zoomSelector.js [hunk 2]
```diff
@@ -18,20 +18,36 @@
...
+
+const StyledDropDown = styled(DropDown)`
+  button {
+    padding: 8px;
+
+    span {
+      padding: 0;
+    }
+  }
+`;
```
Comment (barklund) [inline assets/src/edit-story/components/carousel/zoomSelector/zoomSelector.js]: Sizing and padding aren't correct [according to Figma](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories---Post-Stable?node-id=2241%3A2):

Screen Shot 2021-03-30 at 09 15 02

Note that the zoom dropdown should align with the help center in height and vertical alignment.

Also, there should *not* be a dropdown arrow in this one(?), but perhaps double-check with UX on that one.

Comment (samwhale) [reply assets/src/edit-story/components/carousel/zoomSelector/zoomSelector.js]: > Sizing and padding aren't correct [according to Figma]

I'll fix that right up

> Note that the zoom dropdown should align with the help center in height and vertical alignment.

The height shouldn't be the same according to figma, no?
image
image


> Also, there should _not_ be a dropdown arrow in this one(?), but perhaps double-check with UX on that one.

Already chatted with them - they said they wanted the arrow: https://xwp.slack.com/archives/C01ESBB8ZN1/p1617050704003700

Comment (samwhale) [reply assets/src/edit-story/components/carousel/zoomSelector/zoomSelector.js]: > > Sizing and padding aren't correct [according to Figma]
> 
> I'll fix that right up
> 
> > Note that the zoom dropdown should align with the help center in height and vertical alignment.
> 
> The height shouldn't be the same according to figma, no?

There's inconsistencies in the designs. Have a thread going here:
https://xwp.slack.com/archives/C01ESBB8ZN1/p1617113850007600

Comment (barklund) [reply assets/src/edit-story/components/carousel/zoomSelector/zoomSelector.js]: > The height shouldn't be the same according to figma, no?

`36 === 36`, no?

**Comment by samwhale** [comment]
Chatted with designers:
- should be 8px of margin between buttons
- button should be 32px high
- carat should exist to match other dropdowns

Updated to look like this:
dropdown

File: assets/src/edit-story/components/carousel/zoomSelector/zoomSelector.js [hunk 3]
```diff
@@ -18,20 +18,40 @@
...
+
+  button {
+    height: 32px;
+    padding: 8px;
+    padding-right: 0;
+
+    span {
+      padding: 0;
+    }
+  }
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/carousel/zoomSelector/zoomSelector.js]: if you want - I added `selectButtonStylesOverride` as a prop in `DropDown` that allows CSS to get injected into the `Select` button so you don't have to dig into the dropdown like this if you don't want to.

**Review by BrittanyIRL — APPROVED** [review]
Nice!

File: assets/src/edit-story/components/carousel/zoomSelector/zoomSelector.js (hunk 3 above)
Comment (samwhale) [reply assets/src/edit-story/components/carousel/zoomSelector/zoomSelector.js]: Oh that's dope, I'll use that instead

File: assets/src/edit-story/components/carousel/zoomSelector/zoomSelector.js (hunk 2 above)
Comment (samwhale) [reply assets/src/edit-story/components/carousel/zoomSelector/zoomSelector.js]: Oh I had my dimensions flipped. Long day 🤕 - should be fixed now

**Review by embarks — APPROVED** [review]
Looks good :) all the little details coming together <3

File: assets/src/edit-story/components/carousel/zoomSelector/zoomSelector.js (hunk 2 above)
Comment (barklund) [reply assets/src/edit-story/components/carousel/zoomSelector/zoomSelector.js]: Looks good now - excellent job!

**Review by barklund — APPROVED** [review]

**Pull request merged**